src/makeCalculator.js

```diff
@@ -4,7 +4,31 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add: (currentResult) => (value) => currentResult + value,
+
+    subtract: (currentResult) => (value) => currentResult - value,
+
+    multiply: (currentResult) => (value) => currentResult * value,
+
+    divide: (valueResult) => (value) => {
+      return value === 0 ? 'Error' : valueResult / value;
+    },
+
+    operate: function (callback, value) {
+      this.result = callback(this.result)(value);
+
+      return this;
+    },
+
+    reset: function () {
```

Comment:
```suggestion
    reset() {
```

Reply:
when I write  => function in this place, the tests fail, I think it's related to using "This." Or just tests work that way.

reset: () => {
      this.result = 0;

      return this;
    },